internal/config/endpoints.go

```diff
@@ -14,192 +14,83 @@
-	},
-	"northamerica-northeast1": {
-		URL:        "https://northamerica-northeast1-5tkroniexa-nn.a.run.app",
-		Region:     "northamerica-northeast1",
-		RegionName: "Montréal",
-	},
-	"northamerica-northeast2": {
-		URL:        "https://northamerica-northeast2-5tkroniexa-pd.a.run.app",
-		Region:     "northamerica-northeast2",
-		RegionName: "Toronto",
-	},
-	"southamerica-east1": {
-		URL:        "https://southamerica-east1-5tkroniexa-rj.a.run.app",
-		Region:     "southamerica-east1",
-		RegionName: "São Paulo",
-	},
-	"southamerica-west1": {
-		URL:        "https://southamerica-west1-5tkroniexa-tl.a.run.app",
-		Region:     "southamerica-west1",
-		RegionName: "Santiago",
-	},
-	"us-central1": {
-		URL:        "https://us-central1-5tkroniexa-uc.a.run.app",
-		Region:     "us-central1",
-		RegionName: "Iowa",
-	},
-	"us-east1": {
-		URL:        "https://us-east1-5tkroniexa-ue.a.run.app",
-		Region:     "us-east1",
-		RegionName: "South Carolina",
-	},
-	"us-east4": {
-		URL:        "https://us-east4-5tkroniexa-uk.a.run.app",
-		Region:     "us-east4",
-		RegionName: "North Virginia",
-	},
-	"us-east5": {
-		URL:        "https://us-east5-5tkroniexa-ul.a.run.app",
-		Region:     "us-east5",
-		RegionName: "Columbus",
-	},
-	"us-south1": {
-		URL:        "https://us-south1-5tkroniexa-vp.a.run.app/",
-		Region:     "us-south1",
-		RegionName: "Dallas",
-	},
-	"us-west1": {
-		URL:        "https://us-west1-5tkroniexa-uw.a.run.app",
-		Region:     "us-west1",
-		RegionName: "Oregon",
-	},
-	"us-west2": {
-		URL:        "https://us-west2-5tkroniexa-wl.a.run.app",
-		Region:     "us-west2",
-		RegionName: "Los Angeles",
-	},
-	"us-west3": {
-		URL:        "https://us-west3-5tkroniexa-wm.a.run.app",
-		Region:     "us-west3",
-		RegionName: "Salt Lake City",
-	},
-	"us-west4": {
-		URL:        "https://us-west4-5tkroniexa-wn.a.run.app",
-		Region:     "us-west4",
-		RegionName: "Las Vegas",
-	},
+// GenerateConfigFromEndpoints is used by the cli to generate an Endpoint map
+// using a precompiled list served by the gcping endpoints.
+func GenerateConfigFromEndpoints(ctx context.Context) map[string]Endpoint {
+
+	EndpointsMap := make(map[string]Endpoint)
+
+	req, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodGet,
+		"https://global.gcping.com/api/endpoints",
+		nil,
+	)
+	client := http.DefaultClient
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer resp.Body.Close()
+	json.NewDecoder(resp.Body).Decode(&EndpointsMap)
+
+	return EndpointsMap
+}
+
+// GenerateConfigFromAPI is used to generate the endpoint config through the
+// metadat provided by the Cloud Run Admin API.
+func GenerateConfigFromAPI(ctx context.Context) (map[string]Endpoint, error) {
+	log.Print("Using Cloud Run Admin API to generate Endpoints config.")
+	runService, err := run.NewService(ctx)
+	// TODO: Get project name from Cloud Run metadata service if not defined in env variable
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
```

Comment:
What happens if `projectID` is an empty string?

Reply:
Updated logic to make the code exit with RC=1.

We prefetch the endpoint map and cache at start time, so the container will not start if `projectID` is an empty string, and Cloud Run will keep serving the last working revision.